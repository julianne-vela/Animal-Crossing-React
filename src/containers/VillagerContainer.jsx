import React, { Component } from 'react';
import VillagerDetail from '../components/villagers/VillagerDetail';
import { findVillagerById } from '../services/acnhApi';

export default class VillagerDetails extends Component {
  state = {
    villagerObject: {},
    loading: true,
  };

  async componentDidMount() {
    const villager = await findVillagerById(this.props.match.params.id);
    this.setState({
      villagerObject: villager,
      loading: false,
    });
  }

  render() {
    const { villagerObject, loading } = this.state;

    return (
      <div>
        {loading ? (
          <p>Dodo One, this is Dodo Tower. Do you copy?</p>
        ) : (
          <VillagerDetail villager={villagerObject} />
        )}
      </div>
    );
  }
}
