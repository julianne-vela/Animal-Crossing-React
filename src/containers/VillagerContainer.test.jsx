import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import VillagerContainer from './VillagerContainer';

// snapshot test all presentational components
// behavior test - on changing search refetch articles

describe('Villager Container Test', () => {
  it('should render extended details about the selected villager', async () => {
    render(
      <VillagerContainer
        match={{
          params: {
            _id: '5f5fb4bbbfd05c2aed82e462',
          },
        }}
      />
    );

    const details = await screen.findByRole('region', {
      name: 'villagerDetails',
    });
    expect(details).toMatchSnapshot();
  });
});
