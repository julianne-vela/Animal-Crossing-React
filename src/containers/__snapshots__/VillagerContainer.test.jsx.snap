// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager Container Test should render extended details about the selected villager 1`] = `
<main
  role="villagerDetails"
>
  <label />
  <label />
  <img />
  <ul>
    <li />
    <li />
    <li />
    <li />
    <li />
    <li />
  </ul>
  <a
    href="/"
  >
    Back to List
  </a>
</main>
`;
